test/unit/core/detail/reflection_test.cpp

```diff
@@ -83,7 +85,65 @@ class reflection : public ::testing::Test
 
 TYPED_TEST_CASE(reflection, reflection_types);
 
+TYPED_TEST(reflection, size)
+{
+    EXPECT_EQ(detail::get_display_name_size_v<TypeParam>, this->expected_name().size());
+}
+
 TYPED_TEST(reflection, name)
 {
     EXPECT_EQ(detail::get_display_name_v<TypeParam>.string(), this->expected_name());
 }
+
+namespace seqan3::detail
+{
+template <typename type>
+struct pretty_function
+{
+    static auto as_string()
+    {
+        return std::string{__PRETTY_FUNCTION__};
+    };
+};
+} // namespace seqan3::detail
+
+// NOTE: This does not test a seqan3 library feature, but the underlying magic
+// of how seqan3::detail::get_display_name_v works. This is needed, because
+// __PRETTY_FUNCTION__ has vendor specific output which is not standardised.
+TEST(pretty_function, as_string_detailed)
+{
+    std::string int_ = seqan3::detail::pretty_function<int>::as_string();
+#if defined(__clang__)
+    EXPECT_EQ(int_, "static auto seqan3::detail::pretty_function<int>::as_string() [type = int]");
+#elif defined(__GNUC__)
+    EXPECT_EQ(int_, "static auto seqan3::detail::pretty_function<type>::as_string() [with type = int]");
+#endif
+
+    std::string foo_bar_char = seqan3::detail::pretty_function<foo::bar<char>>::as_string();
+#if defined(__clang__)
+    EXPECT_EQ(foo_bar_char, "static auto seqan3::detail::pretty_function<foo::bar<char> >::as_string() [type = foo::bar<char>]");
+#elif defined(__GNUC__)
+    EXPECT_EQ(foo_bar_char, "static auto seqan3::detail::pretty_function<type>::as_string() [with type = foo::bar<char>]");
+#endif
+}
+
+// NOTE: This does not test a seqan3 library feature, but the underlying magic
+// of how seqan3::detail::get_display_name_v works. This is needed, because
+// __PRETTY_FUNCTION__ has vendor specific output which is not standardised.
+TEST(pretty_function, as_string)
+{
+    // an example how to extract the type
+    auto type_name = [](std::string pretty)
+    {
+        auto start = pretty.find("type = ") + std::string("type = ").size();
+        auto end = pretty.rfind("]");
+
+        return pretty.substr(start, end - start);
+    };
+
+    std::string int_ = seqan3::detail::pretty_function<int>::as_string();
```

Comment:
please use a different name for the `int_` variable. It confused me a second to what it represents.